MoltenVK/MoltenVK/Utility/MVKEnvironment.h

```diff
@@ -286,3 +289,8 @@ void mvkSetConfig(const MVKConfiguration& mvkConfig);
 #ifndef MVK_CONFIG_USE_METAL_ARGUMENT_BUFFERS
 #   define MVK_CONFIG_USE_METAL_ARGUMENT_BUFFERS    MVK_CONFIG_USE_METAL_ARGUMENT_BUFFERS_NEVER
 #endif
+
+/** Compress MSL shader source code in a pipeline cache. Defaults to no compression. */
+#ifndef MVK_CONFIG_SHADER_COMPRESSION_ALGORITHM
+#  	define MVK_CONFIG_SHADER_COMPRESSION_ALGORITHM    MVK_CONFIG_COMPRESSION_ALGORITHM_NONE
```

Comment:
I think it will depend on number of shaders. God of War have 35k shaders that are compiled in menu

Reply:
I tested enabling compression with the zlib algorithm, if you have async shader compilation you will notice a significant stutter.

Reply:
Thanks for the feedback and discussion. 

In #1824, @danginsburg had recommended keeping shader compression below 0.1 ms. From testing using Vulkan CTS  on small shaders (~couple of dozen lines) during development of this PR:
- ZLIB: avg ~0.1 ms and compressed size ~30% of the original size.
- LZ4: avg ~0.04 ms and compressed size ~70% of the original size.

And, no surprise, larger shaders (~couple of hundred lines) compress better (~10% of original), but take longer (~0.4 ms).

I figured it was best to let the app decide to take active action to compress and save memory, if it can absorb the time cost.